pkg/services/authn/authn.go

```diff
@@ -120,6 +129,10 @@ type Service interface {
 	// - "saml" = "auth.client.saml"
 	// - "github" = "auth.client.github"
 	IsClientEnabled(client string) bool
+
+	// GetClientConfig returns the client configuration for the given client.
+	// If the client is not found and/or does not implement the SSOClientConfig, it will return a nil value.
+	GetClientConfig(client string) SSOClientConfig
```

Comment:
I dont like the idea to have the requirement to check for nil value by callers, could we return `(SSOClientConfing, bool)` here instead. That would be a better indicator that the caller must check that everything is ok